bsip-0022.md

```diff
@@ -9,113 +9,439 @@
 
-let max_days_decay = 30; //Set by committee
-let start_decay_date = vote_date()
-let days_since_decay_began = todays_date() - ()
+- `full_power_duration` and `total_decay_duration` are defined in units of a day and shall each have a minimum value of one (1)
+- `total_decay_steps` shall have a minimum value of one (1)
+- `total_decay_duration` in all categories shall be divisible by `total_decay_steps`
 
-if (days_since_decay_began > max_days_decay) {
-    decaying_vote_weight = 0; //Or nullify the vote
-} else {
-    decaying_vote_weight = original_vote_weight - ((original_vote_weight/max_days_decay)*days_since_decay_began)
-}
+**Implementation hints:**
+
+- parameters can be added as child of extensions of the blockchain properties, for example:
+
+```
+parameters.extensions.vote_decay.proxy_delegation.active = true;
+parameters.extensions.vote_decay.workers.full_power_duration = 7776000;
+etc.
 ```
 
-# Discussion
+### Account's Last Voting Time
+
+The [BitShares Core Feature Release of 201902](https://github.com/bitshares/bitshares-core/milestone/18?closed=1) introduced [the recording of an account's last vote time](https://github.com/bitshares/bitshares-core/pull/1449/) and this implementation of this BSIP can re-use it to determine an account's last voting time.
+
+The behavior of that functionality is consistent with the intent of the proposal in that any changes to an account voting slate, even the removal of proxy delegation, is considered an updated an acccount's last voting time.
+
+
+### Calculation of Voting Power
+
+The calculation of voting power is currently performed by the [vote_tally_helper](https://github.com/bitshares/bitshares-core/blob/master/libraries/chain/db_maint.cpp#L1178) during chain maintenance.
 
-## Should the committee have the power to blacklist users from voting?
+It is here that the vote decay could be applied by calculating a voting stake multiplier (&alpha;) which for vote decay would be equal to or less than one (&alpha; &le; 1).  This new function (`calculate_voting_power(opinion_account)`) will calculate the multipler as a function of the current decay parameters and the account's age of last vote.
 
-Whilst exchanges/services voting with user funds (without permission) is bad, such functionality opens the doors for normal users to be included in the blacklist. One would hope that the network would vote such a committee out, but what's to stop the committee then blacklisting users that don't vote for them (or who vote for competitor committee users) so as to maintain power indefinetley? Currently highly unlikely, but still theoretically plausible.
+Any account may have four (4) different multipliers for each of the three (3) referendum categories and the one (1) proxy delegation.  Therefore the `calculate_voting_power(opinion_account)` function can return a multiple-valued structure:
+
+```
+struct voting_power {
+   double for_witness;
```

Comment:
No `double` in consensus code. Each multiplier is a rational value of `1 - (periods passed)/(total periods)`

Reply:
An alternate variable type and pseudo-code was introduced for your review

Reply:
I like your suggestion to use a scaled version of vote tallying. Perhaps best leave the `struct voting_power` completely out of this spec.

Reply:
The `voting_power` struct and the related scaling implementation has been removed